packages/headless/doc-parser/doc-parser.ts

```diff
@@ -1,15 +1,25 @@
 import {ApiModel} from '@microsoft/api-extractor-model';
 import {writeFileSync} from 'fs';
 import {
-  buildControllerConfiguration,
+  ControllerConfiguration,
   resolveController,
 } from './src/controller-resolver';
 
 const apiModel = new ApiModel();
 const apiPackage = apiModel.loadPackage('temp/headless.api.json');
 const entryPoint = apiPackage.entryPoints[0];
 
-const config = buildControllerConfiguration({initializer: 'buildPager'});
+const config: ControllerConfiguration = {
```

Comment:
So we would need to "maintain" this for every new controller ?

I'm trying to think of ways we could have a cookbook or recipe to follow when doing any change in a controller, or when adding new ones, because it's starting to get pretty complicated to remember everything.

The "lots of small work needed to add a new controller" is totally fine by me.
The documentation/sample is extremely important. 

I'm just thinking that maybe we need to start adding a contributor guide in the repo, in the form of a README, which explains everything needed ?

Reply:
> So we would need to "maintain" this for every new controller ?

Yep, this is the config that for a controller will group the initializing function, any utils, the paths to different code samples. 

Every time a new controller is added, one would need to add an config for it. It _may_ be possible to detect and throw an error if a controller is missing a config, I am not sure yet.

For already configured controllers, as the interface of a controller evolves e.g.
- Adds a new property referencing an internal, unexported interface.
- Adds an argument to a function with no description in the tsdoc annotation,

the parser would be able to enforce certain quality checks and throw errors.


We could have a guide, but since some may not read it, I think it's necessary to have a strict post-processing step running on CI that flags and gives helpful error messages. It can also link to the readme.